### ISSUE TITLE
Submitting transaction is refused

### ISSUE THREAD

**Issue by spikeekips** [bosnet/sebak#757]
https://github.com/bosnet/sebak/blob/13dbe0a17626858b51b61b4dde078554b5597458/lib/node/runner/api_node.go#L133-L135

Situation is,

1. Submit the transaction
1. The API response is 200 ok
1. Checking the transaction in history API, 'rejected'
1. The client can not send transaction any more.

@outsideris reported.

**Comment by outsideris** [comment]
@soonkuk The request body is here.

```
{ H:
   { version: '1',
     created: '2018-11-19T18:39:11.832000000+09:00',
     signature:
      'L3yHPqUd5TbzHfxvRCm4Fs9RFKnexizPEz44tDKtNYyLujWJ95J7dYaPnqxiT4HSmV1iXU5ev9jfrcoAEEAT1aF' },
  B:
   { source: 'GCT67AX6PSARM47MJYAEMAQTS4LQRJDJFSMBAEEKHVCDIP5VRC3P2RPC',
     fee: '10000',
     sequence_id: 0,
     operations:
      [ { H: { type: 'congress-voting' },
          B:
           { contract:
              'dHlwZTogbWV0YQoKdGl0bGU6IE1lbWJlcnNoaXAgUmV3YXJkIFBGCgppZDogUEZfUl8wMC0wMDAtQQoKcHJvcG9zZXI6IEJsb2NrY2hhaW5PUyBJbmMKCnByb3Bvc2VyX2FjY291bnQ6IEdCTlVUV1NNNEZSU0VVTFZNSFpGN05GUVdJQkdFREY1WDVPSFhGT1pKQjZTSDVNSUVERUpFSjJGCgpleGVjdXRpb25fZHVyYXRpb246IDYzMDcyMDAKCmFtb3VudF9vZl9pc3N1YW5jZTogMTYwODMzNjAzCgpwZl9idWRnZXRfYWNjb3VudDogR0JXQ01XRFVaSzY3WU5VWjQ0VVBOVkZZWlJTQ0NTNE9MRTZPUldENFpMSTJNVkdZNEtKRFBITU8KCgo=',
             voting: { start: 300, end: 500 },
             funding_address: 'GBWCMWDUZK67YNUZ44UPNVFYZRSCCS4OLE6ORWD4ZLI2MVGY4KJDPHMO',
             amount: '160833603' } } ] } }
```

**Comment by outsideris** [comment]
It was my mistake but it is still a bug.

I set up different congress accounts for each node. It was my mistake. 

```
EROR[11-19|20:45:24] failed to broadcast                      
module=network node=GDIR.M6CC error="{\"code\":163,\"message\":\"http failed to get response\",\"data\":{\"status\":400}}" validator=GAYG.TLY7 type=transaction message=2uTT2gdhftR7iY4Prbh8hMJ2q5nUuraGXpTYUrHjFote response="{\"code\":186,\"message\":\"congress address mismatched\",\"data\":{}}\n" caller=validator_connection_manager.go:214
```

That's why it is not saved in block.

But interesting part is here. When I set up congress voting account is correctly for all node. Congress voting transaction saved in block successfully, but Node 1 throw error `received new, but known message` and node history is `rejected. However Node 2,3 has `confirmed` history. 

I guess a node which receive tx first, receive tx recursively and second request is rejected and save in history.

```
EROR[11-19|20:47:50] failed to broadcast                      module=network node=GDIR.M6CC error="{\"code\":163,\"message\":\"http failed to get response\",\"data\":{\"status\":400}}" validator=GDIR.M6CC type=transaction message=7KEDJqfqhWKMhUcXnGKjVNRRC5H9YbBKSuXn8aQnn6a8 response="{\"code\":106,\"message\":\"received new, but known message\",\"data\":{}}\n" caller=validator_connection_manager.go:214
```

Node 1: 
```
{
  "_links": {
    "account": {
      "href": "/api/v1/accounts/GCT67AX6PSARM47MJYAEMAQTS4LQRJDJFSMBAEEKHVCDIP5VRC3P2RPC"
    },
    "self": {
      "href": "/api/v1/transactions"
    },
    "transaction": {
      "href": "/api/v1/transactions/2uTT2gdhftR7iY4Prbh8hMJ2q5nUuraGXpTYUrHjFote"
    }
  },
  "confirmed": "2018-11-19T20:48:32.152966000+09:00",
  "created": "2018-11-19T20:48:32.142000000+09:00",
  "hash": "2uTT2gdhftR7iY4Prbh8hMJ2q5nUuraGXpTYUrHjFote",
  "source": "GCT67AX6PSARM47MJYAEMAQTS4LQRJDJFSMBAEEKHVCDIP5VRC3P2RPC",
  "status": "rejected"
}
```

Node 2:
```
{
  "_links": {
    "account": {
      "href": "/api/v1/accounts/GCT67AX6PSARM47MJYAEMAQTS4LQRJDJFSMBAEEKHVCDIP5VRC3P2RPC"
    },
    "self": {
      "href": "/api/v1/transactions"
    },
    "transaction": {
      "href": "/api/v1/transactions/2uTT2gdhftR7iY4Prbh8hMJ2q5nUuraGXpTYUrHjFote"
    }
  },
  "confirmed": "2018-11-19T20:48:36.083003000+09:00",
  "created": "2018-11-19T20:48:32.142000000+09:00",
  "hash": "2uTT2gdhftR7iY4Prbh8hMJ2q5nUuraGXpTYUrHjFote",
  "source": "GCT67AX6PSARM47MJYAEMAQTS4LQRJDJFSMBAEEKHVCDIP5VRC3P2RPC",
  "status": "confirmed"
}
```

Node 3:
```
{
  "_links": {
    "account": {
      "href": "/api/v1/accounts/GCT67AX6PSARM47MJYAEMAQTS4LQRJDJFSMBAEEKHVCDIP5VRC3P2RPC"
    },
    "self": {
      "href": "/api/v1/transactions"
    },
    "transaction": {
      "href": "/api/v1/transactions/2uTT2gdhftR7iY4Prbh8hMJ2q5nUuraGXpTYUrHjFote"
    }
  },
  "confirmed": "2018-11-19T20:48:36.082559000+09:00",
  "created": "2018-11-19T20:48:32.142000000+09:00",
  "hash": "2uTT2gdhftR7iY4Prbh8hMJ2q5nUuraGXpTYUrHjFote",
  "source": "GCT67AX6PSARM47MJYAEMAQTS4LQRJDJFSMBAEEKHVCDIP5VRC3P2RPC",
  "status": "confirmed"
}
```

**Comment by spikeekips** [comment]
@outsideris I also think so, that’s why I mentioned the code, that part set the ‘rejected’ in history.

**Comment by outsideris** [comment]
As @soonkuk said, `SaveTransactionHistory` only changed `""` or `submited` state. However, it looks it is changed to `rejected` from `submited`, not `confirmed`.

**Comment by spikeekips** [comment]
@soonkuk saving in history is necessary. To remove this problem, it would be easy way not to save in history. I want to know your opinion.

**Comment by soonkuk** [comment]
I think current problem is like as below.
When node get the transaction.
Transaction is saved in transactionhistory as `submitted`.
But during consensus, transaction is broadcasted to self again and it does not pass validation because it is in transactionhistory as `submitted` state. and It is saved as `rejected`.
This is result of discussion between @outsideris and me.

**Comment by spikeekips** [comment]
@soonkuk tjat’s what I said. My question is, can we remove to save history?

**Comment by soonkuk** [comment]
@spikeekips IMO, node runner's transactionpool and block.transactionpool can prevent duplicated transaction. So I think we can remove to save history.

**Comment by spikeekips** [comment]
Related #760

**Comment by kfangw** [comment]
May I solve this issue by #754 ?

**Comment by spikeekips** [comment]
@kfangw 👍

**Comment by kfangw** [comment]
solved by #768
